Meta/check-headers-acyclic.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python3
```

Comment:
Please reformat this with https://github.com/psf/black. We don't enforce this yet, but I intend to eventually as the number of python scripts in Meta/ is growing.